app/main.py

```
@@ -20,8 +21,51 @@ def sha256_hash_str(to_hash: str) -> str:
     return sha256(to_hash.encode("utf-8")).hexdigest()
 
 
-def brute_force_password() -> None:
-    pass
+TARGETS = set(PASSWORDS_TO_BRUTE_FORCE)
+DIGITS = 8
+TOTAL = 100_000_000
+
+
+def worker_task(args):
+    start, end = args
+    found = {}
+    for i in range(start, end):
+        _string = f"{i:0{DIGITS}d}"
+        _hash = sha256_hash_str(_string)
+        if _hash in TARGETS:
+            found[_hash] = _string
+            if len(found) == len(TARGETS):
+                break
+    return found
+
+
+def brute_force_password(chunk_size: int = 100_000) -> None:
+    processes = max(1, multiprocessing.cpu_count() - 1)
+
+    tasks = []
+
+    for start in range(0, TOTAL, chunk_size):
+        end = min(start + chunk_size, TOTAL)
+        tasks.append((start, end))
+
+    found = {}
+
+    with Pool(processes=processes) as pool:
+        for result in pool.imap_unordered(worker_task, tasks):
+            for _hash, pw in result.items():
+                if _hash not in found:
+                    found[_hash] = pw
+            if len(found) >= len(TARGETS):
+                pool.terminate()
+                break
+
+    assert len(found) == len(TARGETS), \
+        f"Found {len(found)}/{len(TARGETS)} passwords — required all 10."
```

Good: the `assert len(found) == len(TARGETS)` ensures the script fails loudly if not all 10 passwords are discovered. This satisfies the checklist item that all 10 passwords must be found and printed.